templates/demo-store/app/routes/($lang)/$shopid/orders/$token/authenticate.tsx

```diff
@@ -0,0 +1,40 @@
+import {Shop} from '@shopify/hydrogen/storefront-api-types';
+import {redirect, type LoaderArgs} from '@shopify/remix-oxygen';
+import invariant from 'tiny-invariant';
+import {Button, PageHeader} from '~/components';
+
+/*
+ If your online store had active orders before you launched your Hydrogen storefront,
+ and the Hydrogen storefront uses the same domain formerly used by the online store,
+ then customers will receive 404 pages when they click on the old order status URLs
+ that are routing to your Hydrogen storefront. To prevent this, ensure that you redirect
+ those requests back to Shopify.
+*/
+export async function loader({request, context: {storefront}}: LoaderArgs) {
+  const {origin} = new URL(request.url);
+  const {shop} = await storefront.query<{
+    shop: Shop;
+  }>(`query getShopPrimaryDomain { shop { primaryDomain{ url } } }`, {
+    cache: storefront.CacheLong(),
+  });
+  invariant(shop, 'Error redirecting to the order status URL');
+  return redirect(request.url.replace(origin, shop.primaryDomain.url));
```

Comment:
There's a script we recommend merchants adopt in the online store to redirect users to the custom storefront.  If that script isn't implemented correctly, this could create an infinite redirect. Maybe there's a way we could check the request `referer` to prevent a redirect?